src/Cli/dotnet/commands/dotnet-workload/install/FileBasedInstaller.cs

```diff
@@ -348,6 +365,29 @@ public void GarbageCollectInstalledWorkloadPacks(DirectoryPath? offlineCache = n
                     Directory.Delete(packIdDir);
                 }
             }
+
+            if (cleanAllPacks)
+            {
+                EradicateAllWorkloadInstallationRecords(_sdkFeatureBand);
+            }
+        }
+
+        public void EradicateAllWorkloadInstallationRecords(SdkFeatureBand featureBand)
+        {
+            FileBasedInstallationRecordRepository workloadRecordRepository = new(_workloadMetadataDir);
+            var allFeatureBands = workloadRecordRepository.GetFeatureBandsWithInstallationRecords();
+
+            foreach (SdkFeatureBand potentialBandToClean in allFeatureBands)
+            {
+                if (potentialBandToClean.CompareTo(featureBand) < 1)
```

Comment:
As before, maybe we don't need this check at all.

```suggestion
                if (potentialBandToClean.CompareTo(featureBand) < 0)
```